js/matchingView.js

```diff
@@ -0,0 +1,144 @@
+define([
+    'core/js/adapt',
+    'core/js/views/questionView',
+    './dropdownAdapter'
+], function(Adapt, QuestionView, dropdownAdapter) {
+    
+    var MatchingView = QuestionView.extend({
+
+        disableQuestion: function() {
+            this.$('select').prop('disabled', true).select2();
+        },
+
+        setupSelect2: function() {
+            this.enableQuestion();
+            if (this.model.get('_isEnabled') !== true) {
+                // select2 ignores disabled property applied to <select> in the template
+                this.disableQuestion();
+            }
+            _.bindAll(this, 'onOptionSelected');
+            this.$('select').on('select2:select', this.onOptionSelected);
+        },
+
+        enableQuestion: function() {
+            this.$('select').prop('disabled', false).select2({
+                placeholder: this.model.get('placeholder'),
+                minimumResultsForSearch: Infinity, // hides the search box from the Select2 dropdown
+                dir: Adapt.config.get('_defaultDirection'),
+                dropdownAdapter: dropdownAdapter
+            });
+        },
+
+        resetQuestionOnRevisit: function() {
+            this.resetQuestion();
+        },
+
+        setupQuestion: function() {
+            this.listenToOnce(Adapt, 'preRemove', this.onPreRemove);
+
+            this.model.setupRandomisation();
+        },
+
+        onPreRemove: function() {
+            this.$('select').off('select2:select', this.onOptionSelected);
+            this.$('select').select2('destroy');
+        },
+
+        onQuestionRendered: function() {
+            this.setReadyStatus();
+            this.setupSelect2();
```

Comment:
Would it not be better to setup select2 before setting the ready status? May prevent undesirable flickers on rendering

Reply:
Possibly yes. This is how it was before the refactor. I would say let's try swapping the order given that we're checking all this carefully.